src/nvim/message.c

```diff
@@ -241,12 +240,12 @@ msg_strtrunc (
  * Truncate a string "s" to "buf" with cell width "room".
  * "s" and "buf" may be equal.
  */
-void trunc_string(char_u *s, char_u *buf, int room, int buflen)
+void trunc_string(char_u *s, char_u *buf, long room, long buflen)
 {
-  int half;
-  int len;
+  long half;
+  long len;
+  long i;
```

Comment:
Is it a good idea to create a sister function for the ones that takes bad types across the codebase to mitigate the need of casting and eventually replace the old one?